doc/website/blog/2022-05-19-announcing-atlas-projects.md

```diff
@@ -0,0 +1,134 @@
+env "local" {
+  // Declare where the schema definition file resides.
+  src = "./schema/project.hcl"
+  
+  // Define the URL of the database which is managed in
+  // this environment.
+  url = "mysql://localhost:3306"
+  
+  // Define the URL of the Dev Database for this environment
+  // See: https://atlasgo.io/dev-database
+  dev = "mysql://localhost:3307"
+  
+  // The schemas in the database that are managed by Atlas.
+  schemas = ["users", "admin"]
+}
+
+env "dev" {
+  // ... a different env
+}
+```
+
+Project files arose from the need to provide a better experience for developers using the CLI.
+For example, consider you are using Atlas to plan migrations for your database schema. In this case,
+you will be running a command similar to this to plan a migration:
+
+```
+atlas migrate diff --dev-url mysql://root:password@localhost:3306 --to file://schema.hcl --dir file://migrations --format atlas
+```
+
+With project files, you can define an environment named `local`:
+
+```hcl
+env "local" {
+    url = "mysql://root:password@localhost:3306"
+    dev = "mysql://root:password@localhost:3307"
+    src = "./schema.hcl"
+    migration_dir {
+        url = "file://migrations"
+        format = atlas
+    }
+}
+```
+
+Then run the `migrate diff` command against this environment using the `--env` flag:
+
+```
+atlas migrate diff --env local
+```
+
+Alternatively, suppose you want to use Atlas to apply the schema on your staging environment. 
+Without project files, you would use: 
+
+```
+atlas schema apply -u mysql://root:password@db.ariga.dev:3306 --dev-url mysql://root:password@localhost:3307 -f schema.hcl
+```
+To do the same using a project file, define another env named `staging`:
+
+```hcl
+env "staging" {
+  url = "mysql://root:password@db.ariga.dev:3306"
+  dev = "mysql://root:password@localhost:3307"
+  src = "./schema.hcl"
+}
+```
+Then run:
+```
+atlas schema apply --env staging
+```
+
+### Passing credentials as input values
+
+Simlar to [schema definition files](/ddl/sql), project files also support [Input Variables](/ddl/input-variables). This means
+that we can define `variable` blocks on the project file to declare which values should be provided when the file  is 
+evaluated. This mechanism can (and should) be used to avoid committing to source control database credentials. 
+To do this, first define a variable named `db_password`: 
+
+```hcl
+variable "db_password" {
+  type = string
+}
+```
+
+Next, replace the database password in all connection strings with a reference to this variable, for example:
+
+```hcl
+env "staging" {
+  url = "mysql://root:${var.db_password}@db.ariga.dev:3306"
+  dev = "mysql://root:${var.db_password}@localhost:3307"
+  src = "./schema.hcl"
+}
+```
+
+If we run `schema apply` without providing the password input variable, we will receive an 
+error message: 
+
+```
+Error: error reading project file: schemahcl: failed decoding: input value "db_password" expected but missing
```

Comment:
Error message needs improvements. e.g. remove the prefix (not a decode error) and replace "input value" with  "variable" like terraform - e.g. `Error: missing value for required variable "db_password"`.